### ISSUE TITLE
TypeError: Can't instantiate abstract class VisDynamicsModel with abstract methods forward

### ISSUE THREAD

**Issue by fmorenovr** [BoyuanChen/neural-state-variables#4]
Why appears that message?

**Comment by linmx1** [comment]
**I also encountered this problem. How can I solve it? I installed the environment in PyCharm and directly ran the command:**

 python ../main.py ../configs/single_pendulum/model64/config1.yaml
 
 
 {'data_filepath': './data/',
 'dataset': 'single_pendulum',
 'epochs': 1000,
 'gamma': 0.5,
 'if_cuda': True,
 'log_dir': 'logs',
 'lr': 0.001,
 'lr_schedule': [20, 50, 100, 300],
 'model_name': 'encoder-decoder-64',
 'num_gpus': 1,
 'num_workers': 8,
 'seed': 1,
 'test_batch': 256,
 'train_batch': 512,
 'val_batch': 256}
Global seed set to 1

**The error message is as follows:** 
Traceback (most recent call last):
  File "../main.py", line 133, in <module>
    main()
  File "../main.py", line 55, in main
    lr_schedule=cfg.lr_schedule)
  File "C:\Users\linmx\PycharmProjects\pythonProject1\neural-state-variables-main\models.py", line 61, in __init__
    self.__build_model()
  File "C:\Users\linmx\anaconda3\envs\neural_state_variables\lib\site-packages\torch\nn\modules\module.py", line 779, in __getattr__
    type(self).__name__, name))
torch.nn.modules.module.ModuleAttributeError: 'VisDynamicsModel' object has no attribute '_VisDynamicsModel__build_model'

**Comment by BoyuanChen** [comment]
Which version of pytorch-lightning are you using? This looks like a mismatch version of the pytorch-lightning library. Please refer to the requirements.txt file for the version that this code base is built upon: https://github.com/BoyuanChen/neural-state-variables/blob/main/requirements.txt#L130

**Comment by linmx1** [comment]
Still getting errors after trying.

**Comment by linmx1** [comment]
My CUDA version is 12.4
torch 1.10.2+cu113
torchvision 0.11.3+cu113

**When** using pytorch-lightning==1.1.1, it raises the error:
RuntimeError: Distributed package doesn't have NCCL built in
**While** with pytorch-lightning==1.2.5, it raises the error:
TypeError: __init__() got an unexpected keyword argument 'filepath'

It seems that in your code, CUDA version 10.0 does not support the 40 series GPUs. I am planning to replace the keyword data_filepath: './data' in the config1.yaml file with filepath (as well as the corresponding variable mentioned in the text).

Do you have a better approach?
I've been struggling with this for several days now, and I'm requesting your assistance.

**Comment by BoyuanChen** [comment]
Right. The code base was released about two years ago and we did not have a 40 series GPUs back then. I would suggest to look into running it based on our exact suggested configurations which we have tested it, or upgrade packages and GPU version but specifically looking at the newer version of pytorch-lightning documentations. There are some parameter changes on their new APIs. We also plan to release an updated version later this year which will be compatible to newer versions of these packages.
